web/static/js/components/simulation/QuestionnaireSimulation.jsx

```diff
@@ -83,12 +83,30 @@ class QuestionnaireSimulation extends Component<Props, State> {
         })
       }
     }
+
+    // HACK: creates an external function to manually trigger a react route
+    // change from outside react's scope/context where we can't use the Link
+    // component or a callback method
+    window.__refreshSimulation = () => {
+      window.event.preventDefault()
+
+      // HACK: react-router won't navigate to the same URL and doesn't have a
+      // solution to force a refresh... we thus force a refresh by pushing a
+      // fake route, that will unload the current context, and immediately
+      // replace it for the simulator page, that will trigger a refresh.
+      //
+      // See https://dev.to/zbmarius/react-route-refresh-without-page-reload-1907
+      const pathname = window.location.pathname
+      router.push('/')
```

Comment:
Do we have this problem? (from the post)
_But because the app's router was mounted into a nested route, doing history.push('/') un-mounts the entire react app loaded there. This means the whole context gets wiped._

Reply:
@ftarulla This is the actual behavior we want: to wipe the current context, so React will recreate it. Without this hack, React won't reload the context.

There is probably a better solution, maybe call a method that would reset the simulation, instead of playing with the router? If you want to take a stab at it, please do! I'm not confident in my React-fu.

Reply:
Yeah! From what I was reading this is a missing part in `react-router` and all solutions are workarounds/hacks so I don't think we could do much about it.